rust/lazy/src/lazy_rule.rs

```diff
@@ -0,0 +1,142 @@
+use std::{path::Path, str::FromStr};
+
+use crate::error::{LazyError, Result};
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum LazyStrategy {
+    SMUDGE,
+    POINTER,
+}
+
+impl FromStr for LazyStrategy {
+    type Err = LazyError;
+
+    fn from_str(s: &str) -> Result<Self> {
+        let lower_s = s.to_ascii_lowercase();
+        match lower_s.as_str() {
+            "smudge" => Ok(LazyStrategy::SMUDGE),
+            "pointer" => Ok(LazyStrategy::POINTER),
+            "s" => Ok(LazyStrategy::SMUDGE),
+            "p" => Ok(LazyStrategy::POINTER),
+            _ => Err(LazyError::InvalidStrategy(format!(
+                "{s}, supported strategy is smudge(s) or pointer(p)"
+            ))),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct LazyRule {
+    pub path: String,
+    pub strategy: LazyStrategy,
+}
+
+impl FromStr for LazyRule {
+    type Err = LazyError;
+
+    fn from_str(s: &str) -> Result<Self> {
+        let split = s.trim().split_once(' ').ok_or_else(|| {
```

Comment:
This won't support windows with the space here, or filenames containing \ in linux.  I would instead just do an rsplit on ' ' and then use everything to the left as the path.